drivers/amlogic/mmc/aml_sdio.c

```diff
@@ -677,8 +677,8 @@ void aml_sdio_request(struct mmc_host *mmc, struct mmc_request *mrq)
 
     if(mrq->data) {
         /*Copy data to dma buffer for write request*/
-        aml_sdio_prepare_dma(host, mrq);
-        writel(host->bn_dma_buf, host->base + SDIO_ADDR);
+        //aml_sdio_prepare_dma(host, mrq);
+        writel(sg_dma_address(mrq->data->sg), host->base + SDIO_ADDR);
```

Comment:
Where / when mapping between the sg buffer and the DMA / bus address is created? In the mainline driver we use `dma_map_sg()` for this.